src/ingestors/zora-mint/index.ts

```diff
@@ -0,0 +1,102 @@
+import { MintContractOptions, MintIngestor, MintIngestorResources } from '../../lib/types/mint-ingestor';
+import { MintIngestionErrorName, MintIngestorError } from '../../lib/types/mint-ingestor-error';
+import { MintInstructionType, MintTemplate } from '../../lib/types/mint-template';
+import { MintTemplateBuilder } from '../../lib/builder/mint-template-builder';
+import { zoraMintAbi } from './abi';
+import {  getZoraMintPriceInEth,getZoraContractMetadata } from './onchain-zora';
+import { urlForValidZoraPage,zoraOnchainIdDataFromUrl } from './offchain-metadata';
+// import { getProhibitionContractMetadata} from './onchain-zora';
+// import { Address, encodeAbiParameters } from "viem";
+// 
+export class zoraIngestor implements MintIngestor {
+  // async supportsUrl(url: string): Promise<boolean> {
+  //   return new URL(url).hostname === 'zora.co';
+  // }
+  configuration = {
+    supportsContractIsExpensive: true,
+  };
+
+  async supportsUrl(_resources: MintIngestorResources, url: string): Promise<boolean> {
+    return new URL(url).hostname === 'zora.co' ;
+  }
+
+
+  async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+    const { contractAddress,tokenId } = contract;
+    if (!contractAddress || !tokenId) {
+      return false;
+    }
+
+    const url = await urlForValidZoraPage( contractAddress,tokenId, resources.fetcher);
+    return !!url;
+  }
+
+  async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+    const isCompatible = await this.supportsUrl(resources, url);
+    if (!isCompatible) {
+      throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+    }
+
+    const { chainId,tokenId, contractAddress } = await zoraOnchainIdDataFromUrl(url, resources.fetcher);
+
+
+    if (!chainId || !contractAddress || !tokenId) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    return this.createMintForContract(resources, {chainId, contractAddress,tokenId, url });
+  }
+
+
+
+  async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+
+
+
+    const { chainId, contractAddress,tokenId } = contract;
+    if (!chainId || !contractAddress || !tokenId) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('Zora');
+
+    const { name, description, imageUrl,startAt } = await getZoraContractMetadata(Number(tokenId), contractAddress);
+    mintBuilder.setName(name).setDescription(description).setFeaturedImageUrl(imageUrl);
+
+    const totalPriceWei = await getZoraMintPriceInEth(chainId, contractAddress, resources.alchemy);
```

Comment:
I believe we should also be able to get the creator details from Zora, and set them as the `creator`, but... I just realized the MintBuilder doesn't have that field. Will udpate the repo today!

Thanks!